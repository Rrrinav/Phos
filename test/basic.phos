let const GREETING: string = "Hello";
let mutable_value := 10;

let uninitialized: i64;

print(GREETING);
print(mutable_value);

mutable_value = 20;
uninitialized = mutable_value;

print(mutable_value);
print(uninitialized);

{
    let mutable_value := "shadow"; // Shadowing
    print("Inner scope: " + mutable_value);
}

print("Outer scope: " + (mutable_value as string));
