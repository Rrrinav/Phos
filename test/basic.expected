Hello
10
20
20
Inner scope: shadow
Outer scope: 20
